ksqldb-common/src/test/java/io/confluent/ksql/util/timestamp/PartialStringToTimestampParserTest.java

```diff
@@ -129,6 +129,11 @@ public void shouldIncludeRequiredFormatInErrorMessage() {
             + "with an optional numeric 4-digit timezone"));
   }
 
+  @Test
+  public void shouldParseToTimestamp() {
+    assertThat(parser.parseToTimestamp("2017-11-13T23:59:58.999-0100").getTime(), is(1510621198999L));
+  }
+
```

Comment:
The conclusion was to include a conversion function and to not store timezone information. Timezones in strings were not specified in the klip, but I think it makes sense to support it because this is basically the same thing as calling `CONVERT_TZ("2017-11-13T23:59:58.999","-0100")`, and it is something we already support in some filter expressions.